pkg/k8s/utils/utils.go

```diff
@@ -109,6 +109,16 @@ func GetServiceListOptionsModifier() (func(options *v1meta.ListOptions), error)
 	}, nil
 }
 
+// GetLatestPodReadiness returns the lastest podReady condition on a given pod.
+func GetLatestPodReadiness(podStatus slim_corev1.PodStatus) slim_corev1.ConditionStatus {
+	for _, cond := range podStatus.Conditions {
+		if cond.Type == slim_corev1.PodReady {
```

Comment:
If I understand this correctly, we'll return PodReady as soon as we find one of the containers in the pod as ready? Is that desirable? Why don't we check for `ContainersReady` type instead?

```
// ContainersReady indicates whether all containers in the pod are ready.
ContainersReady PodConditionType = "ContainersReady"
```

Reply:
`PodReady` and `ContainerReady` are two different conditions (https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions), the former is more definitive in our context (that the pod is ready for traffic), so we only return the status of `PodReady` condition here.

Reply:
Also, the description for `PodReady` is as follows - 
```
	// PodReady means the pod is able to service requests and should be added to the
	// load balancing pools of all matching services.
	PodReady PodConditionType = "Ready"
```

Based on this, I wonder if the `PodReady` type is meant to be used in the endpoints/autoscaling context. I'm not able to find any clear documentation around which is to be used when.

Reply:
> `PodReady` and `ContainerReady` are two different conditions (https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions), the former is more definitive in our context (that the pod is ready for traffic), so we only return the status of `PodReady` condition here.

Yes, I understand that they are two different conditions, but this explanation is not too convincing for me. If `ContainerReady` is subsumed under `PodReady`, then it makes sense to use the latter. 

>that the pod is ready for traffic

What does this mean? Does this refer to autoscaling like requirements?

Reply:
>  If ContainerReady is subsumed under PodReady, then it makes sense to use the latter.

I think it's the other way around, my understanding is that `PodReady` = `ContainerReady` + readiness probe reporting success.